src/VecSim/algorithms/brute_force/brute_force.cpp

```diff
@@ -199,6 +202,11 @@ VecSimQueryResult_List BruteForceIndex::topKQuery(const void *queryBlob, size_t
         size_t block_size = vectorBlock->getLength();
         vecsim_stl::vector<float> scores(block_size, this->allocator);
         for (size_t i = 0; i < block_size; i++) {
+            if (__builtin_expect(VecSimIndex::timeoutCallback(timeoutCtx), 0)) {
```

Comment:
why not be consistent with the batch iterator method?

Reply:
Not consistent how? in both cases I check the timeout before every distance calculation

Reply:
you created a new function `computeBlockScore`. why not using it here?